src/ChoiceFieldset/ChoiceFieldsetLegend.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import {Box} from '..'
+import VisuallyHidden from '../_VisuallyHidden'
+import {ChoiceFieldsetContext, Slot} from './ChoiceFieldset'
+
+export interface ChoiceFieldsetLegendProps {
+  /**
+   * Whether to visually hide the fieldset legend
+   */
+  visuallyHidden?: boolean
+}
+
+const ChoiceFieldsetLegend: React.FC<ChoiceFieldsetLegendProps> = ({children, visuallyHidden}) => (
+  <Slot name="Legend">
+    {({required, disabled}: ChoiceFieldsetContext) => (
+      <VisuallyHidden
+        as="legend"
+        isVisible={!visuallyHidden}
+        title={required ? 'required field' : undefined}
+        sx={{
+          color: disabled ? 'fg.muted' : undefined,
+          fontSize: 2,
+          padding: 0
+        }}
+      >
+        {required ? (
+          <Box display="flex" as="span">
```

Comment:
It's a wider system props issue perhaps, but shall we use `sx` here instead? This creates some ugly and redundant html.
![Screenshot 2021-12-07 at 12 51 48](https://user-images.githubusercontent.com/13340707/145032336-e2d38885-47ad-4f65-800b-c39f703af535.png)

Reply:
Yea, this looks like an issue with system props. I could use `sx` instead.

@colebemis - is this a known issue?